### ISSUE TITLE
Fatal Error in plugin

### ISSUE THREAD

**Issue by michield** [bramley/phplist-plugin-contentareas#11]
Fatal error: Trait 'phpList\plugin\Common\DAO\MessageTrait' not found in /var/www/phpList3/public_html/lists/admin/plugins/phplist-plugin-common/plugins/Common/DAO/Message.php on line 21

using CommonPlugin 3.21.1+20220915

**Comment by bramley** [comment]
You have somehow installed Common Plugin incorrectly. The path

`/var/www/phpList3/public_html/lists/admin/plugins/phplist-plugin-common/plugins/Common/DAO/Message.php `

should be

`/var/www/phpList3/public_html/lists/admin/plugins/Common/DAO/Message.php `

**Comment by michield** [comment]
Ah, good point. This is when I run it from Git code, and all plugins are in their own repository, instead of in the plugins folder.

https://resources.phplist.com/develop/plugins#multiple_plugin_locations

Is there any way that the plugin can request the code location from the common plugin, and use that instead?

**Comment by bramley** [comment]
Yes I use that approach, like this with full file paths ending in the `plugins` directory
```
define("PLUGIN_ROOTDIRS",
  '/home/duncan/Development/GitHub/phplist-plugin-common/plugins'
. ';/home/duncan/Development/GitHub/phplist-plugin-autoresponder/plugins'

....
```

**Comment by bramley** [comment]
Regardless, I was puzzled as to why the Common/DAO/Message.php  file was found but not Common\DAO\MessageTrait.php

The plugin uses a class map file for the autoloader, and I realised that did not include traits, only classes and interfaces. So traits are loaded using a different method which isn't working in this case.

I have changed the class map to include traits. If you get the latest version of Common Plugin, 3.21.2, then that should fix this problem.
